bin/deploy

```
@@ -0,0 +1,119 @@
+        --application-name="${app}" \
+        --s3-location="s3://${bucket}/${release_name}.zip" \
+        --source=./
+
+    log "Push successful!"
+
+    deploy_config="CodeDeployDefault.OneAtATime"
+    deploy_group="${PREFIX}-explorer-dg${chain_index}"
+
+    log "Creating deployment for CodeDeploy app '${app}', using deployment config ${deploy_config}, and group ${deploy_group}"
+
+    aws create-deployment \
+        --description="${PREFIX} deployment of commit ${CIRCLE_SHA1} via build ${CIRCLE_BUILD_NUM} at ${ts}" \
+        --application-name="${app}" \
+        --deployment-config-name="${deploy_config}" \
+        --deployment-group-name="${deploy_group}" \
+        --s3-location="bucket=${bucket},key=${release_name}.zip,bundleType=zip"
+
+    log 'Deployment successfully created!'
+    log 'You may check deploy progress via the AWS Console or AWS CLI using the deployment ID shown above.'
+
+    return 0
+}
+
+# Make sure we got an environment as an argument to this script
+DEPLOY_ENV="$1"
+if [ -z "$DEPLOY_ENV" ]; then
+    error "Expected deployment environment as an argument, but none was provided!"
+fi
+# Convert it to uppercase for environment variable lookups
+DEPLOY_ENV="$(echo "$DEPLOY_ENV" | tr '[:lower:]' '[:upper:]')"
+
+log "Deploying to environment '${DEPLOY_ENV}'"
+
+# Install AWS CLI
+if ! which aws >/dev/null; then
+    log "Installing AWS CLI.."
+    mkdir -p ~/bin
+    wget "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip"
+    unzip awscli-bundle.zip
+    ./awscli-bundle/install -b ~/bin/aws
+    export PATH=$HOME/bin:$PATH
+    log "Install completed successfully!"
+fi
+
+# Make sure we have credentials set
+log "Verifying credentials.."
+if [ -z "$AWS_ACCESS_KEY_ID" ] || [ -z "$AWS_SECRET_ACCESS_KEY" ]; then
+    error "Unable to perform deploy: missing AWS credentials!"
+fi
+
+# Lookup the infrastructure prefix for the target system in the environment
+log "Verifying infrastructure prefix.."
```

I'd move all these environment variable checks to before the `aws` cli install since they don't depend on install AWS and it will allow misconfigurations to fail faster.